sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -122,14 +126,45 @@ export function convertResponseForQueryBatch(
         return left - right;
       })
       ?.map((response: GeneratedBatchQueryResponse) => ({
-        id: response.id,
         status: response.status,
+        visualization: response.body?.render,
+        statistics: response.body?.statistics,
         // hoist fields from the sub-object 'body' to this level
         error: response.body?.error,
-        tables: response.body?.tables?.map(convertGeneratedTable)
-      }))
+        tables: response.body?.tables?.map((table) => convertGeneratedTable(table)),
+        logsQueryResultStatus: "Success"
+      })),
+    batchResultStatus: "AllSucceeded"
   };
-
+  // compute status for failed or succeed or partial results
+  if (newResponse.results) {
+    let hasError = "Success";
```

Comment:
I don't think `hasError` is the right name for this - `status`? or `batchResultStatus`?

Reply:
this is just used a variable name to compute the logic - called it `resultStatus`